### PR TITLE
FIX : read_raw_edf stim_channel rescaling

### PR DESCRIPTION
**Pull request by alexandrebarachant**
This fix the issue about stim channel discussed on the mailing list.

There was actually two problem :
 * The first one is that offsets where always set to zeros, because of L135 and L136 (tal_channel is usually None, and offsets[None] = 0 nullify the whole array). It was impacting all the channels

* The second one was that the calibration and range were forced to 1 for stim channel.

Here is a zip file containing 10 sec of data and a script to test the patch : https://www.dropbox.com/s/c2yomibdcmdsho8/test_triggers.zip?dl=0

I can add this to the test bench if you want

### CODE REVIEW

**Comment by teonlamont** [comment]
cool. what method did you use to extract the txt. I have a collection of exotic edf/bdf that we usually test against to make sure the modifications are compatible here (edf is a very (maybe too) flexible format). I could share it with you would like. we also have some test files in the module

**Comment by alexandrebarachant** [comment]
For the txt, it was an option of the proprietary software that come with our amplifier (neuroconn). So it is not a conversion from EDF to TXT. Using their software, I did proprietary format => EDF, and proprietary format => TXT.

To be honest, i was a bit surprised to see travis happy, since this patch should definitely change the values of the signal. I guess that it is the 1e-6 that makes the assert_almost_equal okay.

**Comment by agramfort** [comment]
can a test be added or modified? one test that would fail on master but work with this branch?

**Comment by teonlamont** [comment]
the tests we currently do only compares the EEG channels to each other, not the tal/stim channels. this is because the conversion of the TAL channel is handled separately since it is converting to text, and the stim channel specs varied by the system. the stim channel was never scaled by 1e-6

**Comment by agramfort** [comment]
can we improve the tests?

**Comment by teonlamont** [comment]
yes, I think so. in the bdf tests, it can be done.

**Comment by alexandrebarachant** [comment]
I can add the test i provided with the sample data on dropbox.

**Comment by agramfort** [comment]
cool. Please do

**Comment by alexandrebarachant** [comment]
done

**Comment by agramfort** [comment]
if travis and @teonlamont are happy I am happy !

**Comment by Eric89GXL** [comment]
LGTM and Travis is happy as well, @teonlamont feel free to have a look and merge

File: mne/io/edf/tests/test_edf.py [hunk 1]
```diff
@@ -243,4 +246,20 @@ def test_write_annotations():
 
     assert_raises(RuntimeError, read_raw_edf, edf_path, preload=False)
 
+
+def test_edf_stim_channel():
+    """Test stim channel for edf file
+    """
```
Comment (dengemann) [inline mne/io/edf/tests/test_edf.py]: nitpick, the closing """ should be in the same line as the opening ones.

File: mne/io/edf/tests/test_edf.py [hunk 2]
```diff
@@ -243,4 +246,20 @@ def test_write_annotations():
...
+    raw = read_raw_edf(edf_stim_channel_path, preload=True,
+                       stim_channel=-1)
+    true_data = np.loadtxt(edf_txt_stim_channel_path).T
+    # EDF reader pad data if file to small
+    _, Ns = true_data.shape
+    edf_data = raw._data[:, :Ns]
```
Comment (dengemann) [inline mne/io/edf/tests/test_edf.py]: let's avoid uppercase variable names for regular variables

**Comment by dengemann** [comment]
LGTM, besides my 2 pedantic comments.

File: mne/io/edf/tests/test_edf.py (hunk 1 above)
Comment (alexandrebarachant) [inline mne/io/edf/tests/test_edf.py]: I kept the same convention as the other tests in this file.
I updated all of them.

**Comment by agramfort** [comment]
+1 for merge

I'll let @teonlamont merge if is happy. I guess he needs to tests on a few files he has.

thanks @alexandrebarachant

**Comment by teonlamont** [comment]
LGTM. thanks @alexandrebarachant!

**Pull request merged**